ansible/roles/docs-rs-builder/templates/builder.service

```diff
@@ -0,0 +1,25 @@
+#
+# {{ ansible_managed }}
+#
+
+[Unit]
+Description=The docs.rs Builder
+
+[Service]
+Environment=TMPDIR={{ vars_mountpoint_path }}
+Environment=DOCSRS_PREFIX={{ vars_mountpoint_path }}
+Environment=DOCSRS_RUSTWIDE_WORKSPACE={{ vars_mountpoint_path }}
+Environment=DOCSRS_LOG=debug
+Environment=DOCSRS_DATABASE_URL={{ vars_database_url }}
+Environment=DOCSRS_DOCKER_IMAGE=ghcr.io/rust-lang/crates-build-env/linux-micro
+Environment=DOCSRS_STORAGE_BACKEND=s3
+Environment=S3_REGION=us-east-1
+Environment=DOCSRS_S3_BUCKET={{ vars_s3_bucket }}
+Environment=RUST_BACKTRACE=1
+
+WorkingDirectory={{ vars_working_dir }}
+
+ExecStart={{ vars_executable_path }} daemon --registry-watcher=disabled
+
+[Install]
+WantedBy=multi-user.target
```

Comment:
Don't have time right now but I want to compare this to what we're running in prod.

Reply:
Sounds good. We can also land this now and iterate on it as we go forward too.

Reply:
ok, here's the existing config:
```
[Unit]
Description=Cratesfyi daemon
After=network.target postgresql.service

[Service]
User=cratesfyi
Group=cratesfyi
EnvironmentFile=/home/cratesfyi/.docs-rs-env
ExecStart=/home/ubuntu/docs.rs/target/release/cratesfyi daemon
WorkingDirectory=/home/ubuntu/docs.rs
LimitNOFILE=20000
Restart=on-failure

[Install]
WantedBy=multi-user.target
```
The file limit I *think* should be ok to omit because we now upload files as .zips instead of .html, but I'd prefer to keep it around until we have everything else working. `Restart=on-failure` looks important. Environment file is ok to omit since it's configured directly in the .service file, but ideally we'd have some way to configure `DOCSRS_TOOLCHAIN` in prod in case we need to lock it to a specific version.

Here's the env file:
```
DOCSRS_PREFIX=/opt/docs-rs-prefix
DOCSRS_DATABASE_URL=postgresql://cratesfyi@%2Fvar%2Frun%2Fpostgresql
DOCSRS_GITHUB_ACCESSTOKEN=redacted
DOCSRS_RUSTWIDE_WORKSPACE=/home/cratesfyi/workspace
DOCSRS_TOOLCHAIN=nightly
DOCSRS_BUILD_CPU_LIMIT=3
DOCSRS_STORAGE_BACKEND=s3
DOCSRS_LOG=docs_rs=debug,docs_rs::web::strangler=info,rustwide=info
RUST_BACKTRACE=1
SENTRY_DSN=https://redacted@redacted.ingest.sentry.io/redacted
CACHE_CONTROL_MAX_AGE=600
CACHE_CONTROL_STALE_WHILE_REVALIDATE=86400
CLOUDFRONT_DISTRIBUTION_ID_WEB=redacted
CLOUDFRONT_DISTRIBUTION_ID_STATIC=redacted
DOCSRS_CDN_BACKEND=cloudfront
```
All the CACHE_CONTROL vars are important to keep - cc @syphar @jsha to confirm. DOCSRS_CDN_BACKEND and the CLOUDFRONT vars are important to keep. SENTRY_DSN and the GITHUB_ACCESSTOKEN are important. We need some way of configuring secrets, I'm not sure how you're doing that today.

I think it's ok to drop `DOCSRS_BUILD_CPU_LIMIT`, that's set so we can run the web server in parallel but this is just the builder without a web server.

Reply:
Would the build-server only get the config that it needs for the builds? 

Or should the config be the same across the server / container types? 

While it feels cleaner to only have the needed config here, 
due to the shared the codebase it could lead to bugs when we do changes.

Reply:
ah good point, I forgot that CACHE_CONTROL is only used by the web server - ok, those variables are probably unnecessary then. the CLOUDFRONT variables are still necessary since we invalidate the cache after a build, though.

Reply:
> ah good point, I forgot that CACHE_CONTROL is only used by the web server - ok, those variables are probably unnecessary then. the CLOUDFRONT variables are still necessary since we invalidate the cache after a build, though.

That's the kind of questions that made me think if we should have the same config / environment in all server types. At least until we reflect this in our implementation so a developer wouldn't assume locally / in testing that the config is available.

Reply:
The environment variables exposed to the worker vs the web frontend are configured separately in this repo. The web frontend is given its env variables through the terraform config [here](https://github.com/rust-lang/simpleinfra/blob/cb2b9920774a63bd54b3676f2b669ea1e777a91e/terragrunt/modules/docs-rs/web-server.tf#L20-L28) while the worker gets its environment variables through the systemd unit file created through Ansible [here](https://github.com/rust-lang/simpleinfra/pull/231/files#diff-1f5d5ca0e1ea04bafd26697d106e95e5d1d27f2cc4b26e0d7d0138862b46648a). 

It would be pretty difficult to unify these two. I personally think it's best we start treating these two parts of the system separately as they will only become more and more separate over time.

Reply:
I've updated the unit config with the suggestions here except for the CLOUDFRONT variables. I need to understand invalidations a bit better before I can make that change. I personally would prefer we not block merging this PR before we tackle that issue. The point of the staging environment is that we don't need to be perfect before deploying it.

Reply:
👍